spec/views/posts/index_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'capybara/rspec'
+require 'rails_helper'
+
+RSpec.describe 'User profile page', type: :feature do
+  before do
+    @user = User.create(name: 'Tom',
+                        photo: 'https://media.istockphoto.com/id/1289461328/photo/portrait-of-a-handsome-black-man.jpg?s=612x612&w=0&k=20&c=y_mzB0Tbe5LErNy6pqfY7sz2HiDT7fOAUCwupN3-Bg4=',
+                        posts_counter: 5)
+    @posts = [
+      Post.create(author: @user, title: 'Hello',
+                  text: 'This is my first post',
+                  comments: []),
+      Post.create(author: @user, title: 'Greetings',
+                  text: 'I wanted to say hello to everyone!',
+                  comments: [])
+    ]
+  end
+
+  it 'displays the user profile information and posts' do
+    visit user_posts_path(@user)
+
+    expect(page).to have_selector('h2', text: @user.name)
+    expect(page).to have_selector('img[src="https://media.istockphoto.com/id/1289461328/photo/portrait-of-a-handsome-black-man.jpg?s=612x612&w=0&k=20&c=y_mzB0Tbe5LErNy6pqfY7sz2HiDT7fOAUCwupN3-Bg4="]')
+
```

Comment:
- [ ] Your posts index test is missing some few test cases. kindly add the following test cases as demanded by project requirements.
  - The page displays number of posts the user has written
  - The page should display part of the post body
  - Test whether clicking on a post redirects user to the post showpage

Reply:
All tests added. Thanks